mapie/regression/ccp_regression.py

```diff
@@ -0,0 +1,806 @@
+        Fit the estimator and the calibration.
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Training data.
+
+        y: ArrayLike of shape (n_samples,)
+            Training labels.
+
+        sample_weight: Optional[ArrayLike] of shape (n_samples,)
+            Sample weights for fitting the out-of-fold models.
+            If ``None``, then samples are equally weighted.
+            If some weights are null,
+            their corresponding observations are removed
+            before the fitting process and hence have no residuals.
+            If weights are non-uniform, residuals are still uniformly weighted.
+            Note that the sample weight defined are only for the training, not
+            for the calibration procedure.
+
+            By default ``None``.
+
+        groups: Optional[ArrayLike] of shape (n_samples,)
+            Group labels for the samples used while splitting the dataset into
+            train/test set.
+
+            By default ``None``.
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+            By default ``None``
+
+        alpha: Optional[float]
+            Between ``0.0`` and ``1.0``, represents the risk level of the
+            confidence interval.
+            Lower ``alpha`` produce larger (more conservative) prediction
+            intervals.
+            ``alpha`` is the complement of the target coverage level.
+
+            If ``None``, the calibration will be done using the ``alpha``value
+            set in the initialisation. Else, the new value will overwrite the
+            old one.
+
+            By default ``None``
+
+        **fit_params: dict
+            Additional fit parameters for the estimator.
+
+        Returns
+        -------
+        MapieCCPRegressor
+            self
+
+        """
+        self.fit(X, y, sample_weight, groups, **fit_params)
+        self.calibrate(X, y, groups, z, alpha)
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        z: Optional[ArrayLike] = None,
+    ) -> Tuple[NDArray, NDArray]:
+        """
+        Predict target on new samples with confidence intervals.
+        The prediction interval is computed
+
+        Parameters
+        ----------
+        X: ArrayLike of shape (n_samples, n_features)
+            Test data.
+
+        z: Optional[ArrayLike] of shape (n_calib_samples, n_exog_features)
+            Exogenous variables
+
+        Returns
+        -------
+        Tuple[NDArray, NDArray]
+            Tuple[NDArray, NDArray] of shapes (n_samples,)
+            and (n_samples, 2, 1).
+              - [:, 0, 0]: Lower bound of the prediction interval.
+              - [:, 1, 0]: Upper bound of the prediction interval.
+        """
+
+        self.estimator = cast(RegressorMixin, self.estimator)
+        self.conformity_score_ = cast(ConformityScore, self.conformity_score_)
```

Comment:
Compare with `MapieRegressor` how the check is done.

Reply:
Try to add this line:
``check_is_fitted(self, self.fit_attributes + self.calibrate_attributes)``

Reply:
Done. 
Actually, I splitted it into 2, to return only the predictions if ``alpha=None``.


        check_is_fitted(self, self.fit_attributes)
        y_pred = self.estimator_.predict(X)

        if self.alpha is None:
            return y_pred

        check_is_fitted(self, self.calib_attributes)